lib/index.ts

```diff
@@ -179,8 +179,14 @@ export class HUPActionHelper {
 		}
 
 		if (actionName === 'balenahup' && minTakeoverVersion != null) {
+			// OS variant is not relevant and compares less than plain version,
+			// which may erroneously trigger takeover again.
+			const match = currentVersion.match(/\.dev$|\.prod$/);
+			const noVariantVersion = match
+				? currentVersion.substring(0, currentVersion.length - match[0].length)
+				: currentVersion;
```

Comment:
I think you should be able to do v
```suggestion
			const noVariantVersion = currentVersion.replace(/\.dev$|\.prod$/, '');
```